### DISCUSSION TITLE
documents in result webpart are not getting sorted as required using the sort functionality.

### DISCUSSION THREAD

**Discussion by MiteshreeBarolia** [microsoft-search/pnp-modern-search#3456] [General]
We have created customized pnp result webpart and showing the result as the word searched. The result contains documents/data from different lists and libraries. On this when we apply the sort functionality the data is not getting sorted as required. We tried with individual list and library the sort is working fine as expected but when we try to apply sorting when the data is fetched from both list and library the sort functionality is working. 

We are adding the columns for sorting through the managed property and type of property are all Int/date/string. 

Please suggest if there is a way to get the sort to work on all at the same time.

**Comment by kasperbolarsen** [comment]
Can you share the properties of those Managed properties?
They must be sortable in order to work and we often have to map the crawled properties to refinabledate/int/string in order to be able to use they for sorting.

**Comment by MiteshreeBarolia** [comment]
Yes, the properties are sortable and have been mapped to the required respective crawled properties. 

When we try to apply sort on individual list/library it is working, but when we try to apply it when teh result contains both library and list items it does not sort them.
